Intro_Sort/README.md

```diff
@@ -0,0 +1,108 @@
+# Introsort
+
+1. It was invented by David Musser in 1997.
+2. It is also known as introspective sort.
+3. It is a hybrid sorting algorithm(combines two or more other algorithms that solve the same problem,either choosing one depending upon the data or switching between them over the course of algorithm).
+4. It provides both fast average performance and optimal worst performance,thus allowing the performance requirement to become more strict and effective.
+5. It combines the good part of three algorithm i.e quick sort, heap sort,insertion sort.
+
+## ALGORITHM
+
+```
+procedure sort(A : array):
+    let maxdepth = ⌊log(length(A))⌋ × 2
+    introsort(A, maxdepth)
+   
+procedure introsort(A, maxdepth):
+    n ← length(A)
+    // base case
+    if n ≤ 1:
+        return  
+    else if maxdepth = 0:
+        heapsort(A)
+    // assume this function does pivot selection, p is the final position of the pivot
+    else:
+        p ← partition(A)  
+        introsort(A[0:p-1], maxdepth- 1)
+        introsort(A[p+1:n], maxdepth - 1)
+        
+The factor 2 in the maximum depth is arbitrary; it can be tuned for practical performance. A[i:j] denotes the array slice of items i to j.
+```
+
+## Example
+
+![Introsort](https://media.geeksforgeeks.org/wp-content/uploads/20190322164239/page-11.png)
+
+**Input:** 2, 10, 24, 2, 10, 11, 27, 4, 2, 4, 28, 16, 9, 8, 28, 10, 13, 24, 22, 28, 0, 13, 27, 13, 3, 23, 18, 22, 8, 8
+
+**Step 1:** 
+
+​	begin = 0, 	end = 29 	depthLimit = 8, 	a[pivot] = 8. 
+
+As the number of elements > 16,the pivot element is calculated based on the quicksort algorithm. Using the median-of-3 concept,the pivot element is calculated and the a[pivot]=8. The method partition() finds the correct position of the a[pivot] and helps in array split-up for further recursion. Here,the element 8 (in comment) lies in the correct position and in the further steps recursion will be carried over for the sub array before and after 8. 2,2,4,2,4,8,0,3,8,/8/,28,16,9,11,28,10,13,24,22,28,27,13,27,13,24,23,18,22,10,10
+
+
+
+**Step 2:** From the previous step, 
+
+​	begin=0, 	end=29, 	a[pivot]=8 T
+
+he recursion starts for the sub-array a[begin,pivot-1] and a[pivot+1,end] 
+
+2,2,4,2,4,8,0,3,8, 8 ,28,16,9,11,28,10,13,24,22,28,27,13,27,13,27,13,24,23,18,22,10,10 
+
+depthLimit = 7, 	depthLimit = 7, 	begin = 0,	end = 8 	begin = 10,	end = 29 
+
+number of element <= 16 Number of element > 16,
+
+hence sortDataUtil(10,29) that calculates the pivot and a[partition]=27. 
```

Comment:
Space around =, also the first character of new line should begin with Capital letters.